Ical.Net/Calendar.cs

```diff
@@ -187,15 +187,22 @@ public virtual IEnumerable<Occurrence> GetOccurrences(CalDateTime? startTime = n
     public virtual IEnumerable<Occurrence> GetOccurrences<T>(CalDateTime? startTime = null, EvaluationOptions? options = null) where T : IRecurringComponent
     {
         // These are the UID/RECURRENCE-ID combinations that replace other occurrences.
-        var recurrenceIdsAndUids = this.Children.OfType<IRecurrable>()
+        var recurrenceIdsAndUids = Children.OfType<IRecurrable>()
             .Where(r => r.RecurrenceId != null)
-            .Select(r => new { (r as IUniqueComponent)?.Uid, Dt = r.RecurrenceId!.Value })
-            .Where(r => r.Uid != null)
+            // Create a value tuple instead of an anonymous type, because
+            // anonymous types don't work well as dictionary keys due to equality semantics.
+            .Select(r => ((r as IUniqueComponent)?.Uid, r.RecurrenceId!.Value))
+            .Where(x => x.Uid != null)
             .ToDictionary(x => x);
 
         var occurrences = RecurringItems
             .OfType<T>()
-            .Select(recurrable => recurrable.GetOccurrences(startTime, options))
+            .Select(recurrable => recurrable.GetOccurrences(startTime, options)
+                .Where(r =>
+                    (r.Source.RecurrenceId != null) ||
+                    r.Source is not IUniqueComponent uniqueComp ||
+                    !recurrenceIdsAndUids.ContainsKey((uniqueComp.Uid, r.Period.StartTime.Value)))
+            )
```

Comment:
Maybe I suggest you add a code comment here to explain the order/behavior?

Please fill the `.......`

```suggestion
            // Filter UID/RECURRENCE-ID combinations that replace other occurrences before using `.OrderedDistinct()` to ensure that ......
            .Where(r =>
                (r.Source.RecurrenceId != null) ||
                r.Source is not IUniqueComponent uniqueComp ||
                !recurrenceIdsAndUids.ContainsKey((uniqueComp.Uid, r.Period.StartTime.Value))
              )
            )
```